adventure/role.py

```diff
@@ -0,0 +1,322 @@
+from datetime import datetime, timezone
+import logging
+from typing import Optional
+
+import discord
+from discord.ext import tasks
+from discord.ext.commands.cooldowns import BucketType
+from discord.role import Role
+from redbot.core import Config, commands
+from redbot.core.bot import Red
+from redbot.core.commands import Context
+from redbot.core.i18n import Translator, cog_i18n
+
+from .charsheet import parse_timedelta
+from .utils import smart_embed
+
+_ = Translator("Adventure", __file__)
+
+log = logging.getLogger("red.cogs.adventure")
+
+@cog_i18n(_)
+class RoleMixin(commands.Cog):
+    def __init__(self, bot: Red):
+        self.bot = bot
+
+        self.config: Config
+
+    @commands.group(name="roleset")
+    @commands.guild_only()
+    @commands.admin_or_permissions(manage_guild=True)
+    async def _roleset(self, ctx: Context):
+        """Set roles for adventure pings."""
+
+    @_roleset.command(name="general")
+    async def _roleset_general(self, ctx: Context, *, role: Optional[discord.Role]):
+        """Set role for all adventure pings."""
+
+        await self.config.guild(ctx.guild).general_ping_role.set(getattr(role, "id", None))
+
+        if not role:
+            await smart_embed(ctx, _("Unset all adventures role."), True)
+        else:
+            await smart_embed(
+                ctx,
+                _("Set {role} as all adventures role.".format(role=role.mention)),
```

Comment:
```suggestion
                _("Set {role} as all adventures role.").format(role=role.mention),
```
Wrong bracket placement with the translator